app.go

```diff
@@ -943,3 +966,29 @@ func (app *App) printTopic(buffer string) {
 		Body:      ui.Styled(body, tcell.StyleDefault.Foreground(tcell.ColorGray)),
 	})
 }
+
+func getLastBufferPath() string {
+	// TODO make this clean
+	// how to handle flag?
```

Comment:
Please add `func (*App) currentBuffer() string` and move the whole file reading/writing logic to `cmd/senpai/main.go`, which has the path to the configuration file (but it's completely unnecessary for this since the cache dir is more appropriate as simon said).

In `cmd/senpai/main.go`:

1. get `os.UserCacheDir`, or `~/.cache`, the same way it's done for `os.UserConfigDir`
2. `os.MkDirAll` `~/.cache/senpai` since it might not exist
3. read `lastbuffer.txt` and pass it to `NewApp` as another argument next to `cfg`
4. on `app.Close`, write the current buffer back